circuit-benchmarks/src/bytecode_circuit.rs

```diff
@@ -0,0 +1,142 @@
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06,
+            0xbc, 0xe5,
+        ]);
+
+        // Bench setup generation
+        let setup_message = format!("Setup generation with degree = {}", degree);
+        let start1 = start_timer!(|| setup_message);
+        let general_params = ParamsKZG::<Bn256>::setup(degree, &mut rng);
+        let verifier_params: ParamsVerifierKZG<Bn256> = general_params.verifier_params().clone();
+        end_timer!(start1);
+
+        // Initialize the proving key
+        let vk = keygen_vk(&general_params, &bytecode_circuit).expect("keygen_vk should not fail");
+        let pk =
+            keygen_pk(&general_params, vk, &bytecode_circuit).expect("keygen_pk should not fail");
+        // Create a proof
+        let mut transcript = Blake2bWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+
+        // Bench proof generation time
+        let proof_message = format!("Bytecode Proof generation with {} rows", degree);
+        let start2 = start_timer!(|| proof_message);
+        create_proof::<
+            KZGCommitmentScheme<Bn256>,
+            ProverSHPLONK<'_, Bn256>,
+            Challenge255<G1Affine>,
+            XorShiftRng,
+            Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>>,
+            BytecodeCircuitTester<Fr>,
+        >(
+            &general_params,
+            &pk,
+            &[bytecode_circuit],
+            &[&[instance.as_slice()][..]][..],
+            rng,
+            &mut transcript,
+        )
+        .expect("proof generation should not fail");
+        let proof = transcript.finalize();
+        end_timer!(start2);
+
+        // Bench verification time
+        let start3 = start_timer!(|| "Bytecode Proof verification");
+        let mut verifier_transcript = Blake2bRead::<_, G1Affine, Challenge255<_>>::init(&proof[..]);
+        let strategy = SingleStrategy::new(&general_params);
+
+        verify_proof::<
+            KZGCommitmentScheme<Bn256>,
+            VerifierSHPLONK<'_, Bn256>,
+            Challenge255<G1Affine>,
+            Blake2bRead<&[u8], G1Affine, Challenge255<G1Affine>>,
+            SingleStrategy<'_, Bn256>,
+        >(
+            &verifier_params,
+            pk.get_vk(),
+            strategy,
+            &[&[instance.as_slice()][..]][..],
+            &mut verifier_transcript,
+        )
+        .expect("failed to verify bench circuit");
+        end_timer!(start3);
+    }
+
+    /// fill bytecodes_num * bytecode_len bytes to the witness table
+    fn fillup_codebytes<F: Field>(
+        bytecodes_num: usize,
+        bytecode_len: usize,
+        randomness: F,
+    ) -> Vec<UnrolledBytecode<F>> {
+        if bytecodes_num == 0 {
+            vec![]
+        } else {
+            assert!(bytecode_len >= 1);
+            (0..bytecodes_num * bytecode_len)
+                .collect::<Vec<usize>>()
+                .chunks(bytecode_len)
+                .map(|idx_range| {
```

Comment:
Simple `for` loop makes ,more sense here? Because you do `bytecodes_num * bytecode_len` and then create chunks of `bytecode_len`, so may as well just use some loops?

Reply:
Yes, only diff is the value sequence, changed to for_each loop and make the fillup logic simpler.